flake.nix

```diff
@@ -162,49 +205,56 @@
             pkgs.stdenv.mkDerivation {
               inherit pname version src;
 
-              nativeBuildInputs = [effektBuild pkgs.gnused];
-              buildInputs = buildInputs ++ pkgs.lib.concatMap (b: b.buildInputs) backends;
+              # Build-time only dependencies
+              nativeBuildInputs = [
+                effektBuild
+                pkgs.gnused
+                pkgs.makeWrapper
+              ] ++ pkgs.lib.concatMap (b: b.buildInputs) backends;
+
+              # Runtime dependencies for the build environment (needed for tests)
+              buildInputs = buildInputs
+                ++ pkgs.lib.concatMap (b: b.runtimeInputs) backends;
 
-              # TODO: consider removing the 'js-web'-related hacks...
               buildPhase = ''
                 mkdir -p out
 
                 ${pkgs.lib.concatMapStrings (backend: ''
                   echo "Building with backend ${backend.name} file ${src}/${main}"
-                  echo "Current directory: $(pwd)"
-                  echo "Contents of current directory:"
-                  ls -R
                   effekt --build --backend ${backend.name} ${src}/${main}
 
-                  echo "Contents of out directory:"
-                  ls -R out/
-
-                  if [ "${backend.name}" = "js-web" ]; then
-                    echo "Moving .js and .html for js-web backend"
-                    mv "out/$(basename ${src}/${main} .effekt).js" out/${pname}.js
-                    mv "out/$(basename ${src}/${main} .effekt).html" out/${pname}.html
-                    sed -i 's/src="main.js"/src="${pname}.js"/' out/${pname}.html
-                  else
-                    mv out/$(basename ${src}/${main} .effekt) out/${pname}-${backend.name}
-                  fi
+                  ${backend.processOutput pname backend "${src}/${main}"}
+
+                  ${if backend.runtime != null then ''
+                    echo "Setting runtime to ${backend.runtime}"
+                    sed -i '1c#!/usr/bin/env ${backend.runtime}' out/${pname}-${backend.outputName}
+                  '' else ""}
```

Comment:
I'm not confident about this, I'm not sure this works well for Chez...

Reply:
Yeah, we shouldn't do this.
But what should we actually do for `bun`? What's the correct shebang, esp. in a Nix context?